### ISSUE TITLE
`AllCops: Exclude` from a remote config is ignored when `--config` is specified

### ISSUE THREAD

**Issue by wai-doi** [rubocop/rubocop#14745]
## Steps to reproduce the problem

Prepare a `rubocop.yml` file at a remote URL with an `AllCops: Exclude` setting.

```yml
# remote-rubocop.yml
AllCops:
  Exclude:
    - "tmp/**/*"
```

Load this file from the local `.rubocop.yml`.

```yml
inherit_from:
  - https://example.com/path/to/remote-rubocop.yml

AllCops:
  NewCops: enable
```

Create a Ruby file under the excluded directory.

```
$ mkdir tmp
$ touch tmp/test.rb
```

Run RuboCop against the excluded directory while explicitly specifying the config file with `--config`.

```
$ bundle exec rubocop tmp --config .rubocop.yml
Inspecting 1 file
W

Offenses:

tmp/test.rb:1:1: W: Lint/EmptyFile: Empty file detected.

1 file inspected, 1 offense detected
```

## Expected behavior

The excluded directory should not be inspected, and no offenses should be detected.

```
$ bundle exec rubocop tmp --config .rubocop.yml
Inspecting 0 files


0 files inspected, no offenses detected
```

## Actual behavior

When the `--config` option is used to explicitly specify `.rubocop.yml`, the `AllCops: Exclude` setting loaded from the remote URL does not take effect, and files under the excluded directory are inspected.

```
$ bundle exec rubocop tmp --config .rubocop.yml --debug
For /Users/yusuke.doi/sandbox/rubocop-sandbox: configuration from /Users/yusuke.doi/sandbox/rubocop-sandbox/.rubocop.yml
configuration from /Users/yusuke.doi/sandbox/rubocop-sandbox/.rubocop-remote-f4efa56586eb60344b3fd89424e6b7c1.yml
Default configuration from /Users/yusuke.doi/.rbenv/versions/3.4.8/lib/ruby/gems/3.4.0/gems/rubocop-1.82.0/config/default.yml
configuration from /Users/yusuke.doi/sandbox/rubocop-sandbox/.rubocop.yml
configuration from /Users/yusuke.doi/.cache/rubocop_cache/.rubocop-remote-f4efa56586eb60344b3fd89424e6b7c1.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/yusuke.doi/sandbox/rubocop-sandbox/tmp/test.rb
Loading cache from /Users/yusuke.doi/.cache/rubocop_cache/4d3d19306e462e0886f959a6d1b991c1b742f014/a95122e0c77ada5f35b1a0582561d490196ad279/058c691e430652027c9a56979ebdad2257e1611a
W

Offenses:

tmp/test.rb:1:1: W: Lint/EmptyFile: Empty file detected.

1 file inspected, 1 offense detected
Finished in 0.054102999998576706 seconds
```

However, when RuboCop is run without the `--config` option, the `AllCops: Exclude` setting from the remote config works as expected.

```
$ bundle exec rubocop tmp
Inspecting 0 files


0 files inspected, no offenses detected
```

This issue does not occur with 1.81.7, but occurs with 1.82.0.

## RuboCop version

```
$ bundle exec rubocop -V
1.82.0 (using Parser 3.3.10.0, Prism 1.7.0, rubocop-ast 1.48.0, analyzing as Ruby 3.4, running on ruby 3.4.8) [arm64-darwin24]
```

**Comment by wai-doi** [comment]
This issue is expected to be resolved by the changes in PR #14752.

**Comment by koic** [comment]
@wai-doi #14752 has been merged into the master branch. Can you confirm whether the issue is resolved by, for example, specifying `gem 'rubocop', github: 'rubocop/rubocop'` in your Gemfile?

**Comment by wai-doi** [comment]
I’ve confirmed that the issue is resolved.

```
$ cat Gemfile
# frozen_string_literal: true

source 'https://rubygems.org'

gem 'rubocop', github: 'rubocop/rubocop'

$ bundle exec rubocop tmp --config .rubocop.yml
Inspecting 0 files


0 files inspected, no offenses detected
```

Thank you!
